Add ServiceParameter support to the system resource

albailey opened the pull request:
- Update to the latest gophercloud.
The latest gophercloud added support for serviceparameters api.

 - Adds a new serviceParameters section to System spec.
This allows service parameters to be configured during
deployment-manager zero-touch-provisioning.

Example of updating an existing service parameter (the default was 300)

apiVersion: starlingx.windriver.com/v1
kind: System
spec:
  serviceParameters:
  - service: horizon
    section: auth
    paramname: lockout_seconds
    paramvalue: "333"

Example of adding an service parameter (note: the resource field is required)

apiVersion: starlingx.windriver.com/v1
kind: System
spec:
  serviceParameters:
  - service: platform
    section: docker
    paramname: docker_cleanup_cron_hour
    paramvalue: "20"
    resource: "platform::docker::params::docker_cleanup_cron_hour"

Signed-off-by: albailey <Al.Bailey@windriver.com>

donpenney commented on pkg/controller/system/system_controller.go:
```diff
@@ -347,6 +348,102 @@ func (r *ReconcileSystem) ReconcilePTP(client *gophercloud.ServiceClient, instan
+		// If a service parameter is found that does not exist in Info, need to create it
+		// Note: the sysinv API does not actually support creating a new service param.  It will be rejected.
+		if !found {
+			// TODO: populate this params object
```
Is there something more to be done here?

albailey replied:
No.   I did eventually populate the object, but forgot to remove my todo.  I will remove the todo.

mpeters-wrs commented on config/crds/starlingx_v1_system.yaml:
```diff
@@ -149,6 +149,46 @@ spec:
+                    type: string
+                  service:
+                    description: Service identifies the service for this service parameter
+                    maxLength: 255
```
Maximum length is 16.

albailey replied:
OK, I'll change that

mpeters-wrs commented on config/crds/starlingx_v1_system.yaml:
```diff
@@ -149,6 +149,46 @@ spec:
+                    type: string
+                  section:
+                    description: Section identifies the section for this service parameter
+                    maxLength: 255
```
Maximum length is 128

mpeters-wrs commented on pkg/apis/starlingx/v1/system_types.go:
```diff
@@ -91,6 +91,41 @@ func (in *LicenseInfo) DeepEqual(other *LicenseInfo) bool {
+type ServiceParameterInfo struct {
+	// Service identifies the service for this service parameter
+	// +kubebuilder:validation:Pattern=^[a-zA-Z0-9\-_]+$
+	// +kubebuilder:validation:MaxLength=255
```
MaxLength=16

mpeters-wrs commented on pkg/apis/starlingx/v1/system_types.go:
```diff
@@ -91,6 +91,41 @@ func (in *LicenseInfo) DeepEqual(other *LicenseInfo) bool {
+
+	// Section identifies the section for this service parameter
+	// +kubebuilder:validation:Pattern=^[a-zA-Z0-9\-_]+$
+	// +kubebuilder:validation:MaxLength=255
```
MaxLength=128

mpeters-wrs commented on pkg/apis/starlingx/v1/system_types.go:
```diff
@@ -91,6 +91,41 @@ func (in *LicenseInfo) DeepEqual(other *LicenseInfo) bool {
+	ParamName string `json:"paramname"`
+
+	// ParamValue identifies the value for this service parameter
+	ParamValue string `json:"paramvalue"`
```
MaxLength=4096

albailey replied:
updated

mpeters-wrs commented on config/crds/starlingx_v1_system.yaml:
```diff
@@ -149,6 +149,46 @@ spec:
+                    maxLength: 255
+                    pattern: ^[a-zA-Z0-9\-_]+$
+                    type: string
+                  paramvalue:
```
Maximum length is 4096

albailey replied:
updated

mpeters-wrs commented on pkg/controller/system/system_controller.go:
```diff
@@ -347,6 +348,102 @@ func (r *ReconcileSystem) ReconcilePTP(client *gophercloud.ServiceClient, instan
+			}
+		}
+		// If a service parameter is found that does not exist in Info, need to create it
+		// Note: the sysinv API does not actually support creating a new service param.  It will be rejected.
```
I think this comment note is invalid and should be removed.

albailey replied:
will do

albailey commented:
I have updated the max length for param_value to 4096 
I have updated section to 128, and service to 16
I also added maxlength for personality and resource based on the model.py and the migration scripts in sysinv to be 255
